src/daft-core/src/array/ops/utf8.rs

```diff
@@ -299,72 +332,60 @@ impl Utf8Array {
     }
 
     pub fn split(&self, pattern: &Utf8Array, regex: bool) -> DaftResult<ListArray> {
-        let self_arrow = self.as_arrow();
-        let pattern_arrow = pattern.as_arrow();
-        // Handle all-null cases.
-        if self_arrow
-            .validity()
-            .map_or(false, |v| v.unset_bits() == v.len())
-            || pattern_arrow
-                .validity()
-                .map_or(false, |v| v.unset_bits() == v.len())
-        {
-            return Ok(ListArray::full_null(
-                self.name(),
-                &DataType::List(Box::new(DataType::Utf8)),
-                std::cmp::max(self.len(), pattern.len()),
-            ));
-        // Handle empty data and pattern case.
-        } else if self.is_empty() && pattern.is_empty() {
-            return Ok(ListArray::empty(
-                self.name(),
-                &DataType::List(Box::new(DataType::Utf8)),
-            ));
+        let is_valid = check_valid_inputs(
+            self,
+            &[pattern],
+            "split",
+            &DataType::List(Box::new(DataType::Utf8)),
+        );
+        if let Some(result) = is_valid {
+            return result;
         }
-        let buffer_len = self_arrow.values().len();
 
+        let result_len = std::cmp::max(self.len(), pattern.len());
```

Comment:
We can use info returned from the `check_valid_inputs` call to do this, instead of duplicating logic in the caller here.